src/gluonts/core/ty.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import functools
+import inspect
+from typing import Any, Optional
+
+import pydantic
+
+
+def get_param_type(param):
+    if param.annotation == inspect.Parameter.empty:
+        return Any
+
+    return param.annotation
+
+
+def get_param_default(param):
+    if param.default == inspect.Parameter.empty:
+        return ...
```

Comment:
what is this?

Reply:
Pydantic. I've copied this logic from `validated`.

Reply:
See: https://pydantic-docs.helpmanual.io/usage/models/#required-fields